src/components/IndexTable/IndexTable.tsx

```diff
@@ -197,7 +250,16 @@ function IndexTableBase({
     scrollingContainer.current = false;
   }, []);
 
-  useEffect(() => resizeTableHeadings(), [headings, resizeTableHeadings]);
+  useEffect(() => {
+    tableHeadings.current = getTableHeadings();
```

Comment:
Curious to get other input on this; I'm setting `tableHeadings` / `stickyTableHeadings` on a `ref` instead of in a `state` var since we don't need to do a re-render after it happens; as long as we set it in this `useEffect` we're covered whenever it's used by the resizer.